### PR TITLE
Add CORS to dotnet backend API

### PR DESCRIPTION
**Pull request by MortFred**
Also updated the AzureAD resources to allow for this authentication flow.
Backend registration: https://portal.azure.com/#blade/Microsoft_AAD_RegisteredApps/ApplicationMenuBlade/Overview/appId/ea4c7b92-47b3-45fb-bd25-a8070f0c495c/isMSAApp/
Frontend registration: https://portal.azure.com/#blade/Microsoft_AAD_RegisteredApps/ApplicationMenuBlade/Overview/appId/f5993820-b7e2-4791-886f-f9f5027dc7be/isMSAApp/

### CODE REVIEW

File: backend/api/Program.cs [hunk 1]
```diff
@@ -45,6 +45,7 @@
         .Build();
 });
 
+
```
Comment (aestene) [inline backend/api/Program.cs:48]:
```suggestion
```

**Review by aestene — CHANGES_REQUESTED** [review]
Looks good! You should rebase on main before you're able to merge.

It may be that you have to run csharpier too depending on your code.

**Review by GodVenn — COMMENTED** [review]
Nice!
Just one question about naming.

**Comment by GodVenn** [comment]
Also, why is there still a frontend app registration?
Didn't we conclude that there should be only one?

File: frontend/src/authConfig.tsx [hunk 2]
```diff
@@ -13,12 +13,16 @@ export const msalConfig = {
     }
 };
 
+export const APIResourceIDs = {
+    backendAPIID: "ea4c7b92-47b3-45fb-bd25-a8070f0c495c"
```
[the remark this answers is not in the document]
Comment (MortFred) [reply frontend/src/authConfig.tsx:17]: If we need to call several APIs from the frontend, we need to store their AzureAD client id somewhere

**Comment by MortFred** [comment]
> Also, why is there still a frontend app registration? Didn't we conclude that there should be only one?

We need two, we found out that we cannot have a SPA and a webAPI be registered on the same AzureAD identity.

File: frontend/src/authConfig.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (GodVenn) [reply frontend/src/authConfig.tsx:17]: I think we should only call one API from the frontend - our backend.
The backend is responsible for all further calls to other API's and services

**Comment by GodVenn** [comment]
> > Also, why is there still a frontend app registration? Didn't we conclude that there should be only one?
> 
> We need two, we found out that we cannot have a SPA and a webAPI be registered on the same AzureAD identity.

And do we need both?
Our app is a single identity with a single authorization layer. This is the reason for why I strongly believe we should only have one app registration for one app.
The frontend and backend do not have separate access controls.

**Review by GodVenn — APPROVED** [review]
LGTM

**Review by aestene — APPROVED** [review]
LGTM

**Pull request merged**